python_src/src/lamp_py/aws/s3.py

```diff
@@ -349,26 +350,34 @@ def write_parquet_file(
     )
     process_logger.log_start()
 
-    # generate partitioning for this table write based on what columns
-    # we expect to be able to partition out for this input type
-    partitioning = ds.partitioning(
-        table.select(partition_cols).schema, flavor="hive"
-    )
+    # pull out the partition information into a list of strings.
+    partition_strings = []
+    for col in partition_cols:
+        unique_list = pc.unique(table.column(col)).to_pylist()
+        assert len(unique_list) == 1
+        partition_strings.append(f"{col}={unique_list[0]}")
+
+    table = table.drop(partition_cols)
 
+    # generate an s3 path to write this file to
     if basename_template is None:
         basename_template = guid() + "-{i}.parquet"
-
-    ds.write_dataset(
-        data=table,
-        base_dir=s3_dir,
-        filesystem=fs.S3FileSystem(),
-        format=ds.ParquetFileFormat(),
-        partitioning=partitioning,
-        file_visitor=visitor_func,
-        basename_template=basename_template,
-        existing_data_behavior="overwrite_or_ignore",
+    write_path = os.path.join(
+        s3_dir, *partition_strings, basename_template.format(i=0)
```

Comment:
question/nitpick:

I think `write_dataset` required us to have the `{i}` notation in the `basename_template` but that isn't really required anymore and we're not using the `basename_template` parameter for it's true intended purpose, so I'm wondering if it might be worth dropping/renaming that parameter along with the `{i}` notation.

Reply:
i had that thought as well. i left it like this just to keep things consistent with the prior behavior.